src/generator/AutoRest.Go/Extensions.cs

```diff
@@ -904,9 +904,9 @@ public static bool IsBodyParameter(this IParameter p)
         /// <returns></returns>
         public static string GetConstraint(string name, string constraintName, string constraintValue, bool chain = false)
         {
-            var value = constraintName == Constraint.Pattern.ToString()
-                                             ? string.Format("\"{0}\"", constraintValue.Replace("\\", "\\\\"))
-                                             : constraintValue;
+            var value = constraintName == Constraint.Pattern.ToString()  
```

Comment:
minor: if you know constraintName going to be string then i'd prefer string instead of var.